lib/builders/graviton-builder.ts

```diff
@@ -0,0 +1,176 @@
+        );
+    }
+
+    public addIstioControlPlaneAddOn(props?: addons.IstioControlPlaneAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.IstioControlPlaneAddOn(props)
+        );
+    }
+
+    public addKubeStateMetricsAddOn(props?: addons.KubeStateMetricsAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.KubeStateMetricsAddOn(props)
+        );
+    }
+
+    public addMetricsServerAddOn(props?: addons.MetricsServerAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.MetricsServerAddOn(props)
+        );
+    }
+
+    public addPrometheusNodeExporterAddOn(props?: addons.PrometheusNodeExporterAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.PrometheusNodeExporterAddOn(props)
+        );
+    }
+
+    public addExternalsSecretsAddOn(props?: addons.ExternalsSecretsAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.ExternalsSecretsAddOn(props)
+        );
+    }
+
+    public addSecretsStoreAddOn(props?: addons.SecretsStoreAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.SecretsStoreAddOn(props)
+        );
+    }
+
+    public addCalicoOperatorAddOn(props?: addons.CalicoOperatorAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.CalicoOperatorAddOn(props)
+        );
+    }
+
+    public addCertManagerAddOn(props?: addons.CertManagerAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.CertManagerAddOn(props)
+        );
+    }
+
+    public addAdotCollectorAddOn(props?: addons.AdotCollectorAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.AdotCollectorAddOn(props)
+        );
+    }
+
+    public addAmpAddOn(props: addons.AmpAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.AmpAddOn(props)
+        );
+    }
+
+    public addCloudWatchLogsAddOn(props: addons.CloudWatchLogsAddonProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.CloudWatchLogsAddon(props)
+        );
+    }
+
+    public addEfsCsiDriverAddOn(props?: addons.EfsCsiDriverProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.EfsCsiDriverAddOn(props)
+        );
+    } 
+
+    public addFluxCDAddOn(props?: addons.FluxCDAddOnProps) : GravitonBuilder {  
+        return this.addOns(
+            new addons.FluxCDAddOn(props)
+        );
+    }  
+
+    public addGrafanaOperatorAddOn(props?: addons.GrafanaOperatorAddonProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.GrafanaOperatorAddon(props)
+        );
+    }  
+
+    public addXrayAdotAddOn(props?: addons.XrayAdotAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.XrayAdotAddOn(props)
+        );
+    }
+
+    public addOns(...addOns: spi.ClusterAddOn[]): this {
```

Comment:
If we have this method we don't need explicit methods for the addons, like addIstio, etc. If they add an addon that does not support ARM they will get an error telling them. We have docs for reference.